data/pigui/game.lua

```diff
@@ -0,0 +1,539 @@
+	local player = Game.player
+	if target then
+		local velocity = player:GetVelocityRelTo(target)
+		local position = player:GetPositionRelTo(target)
+
+		local uiPos = ui.pointOnClock(center, radius, 2)
+		-- label of target
+		local tooltip = reticuleTarget == "combatTarget" and lui.HUD_CURRENT_COMBAT_TARGET or (reticuleTarget == "navTarget" and lui.HUD_CURRENT_NAV_TARGET or lui.HUD_CURRENT_FRAME)
+		ui.addStyledText(uiPos, target.label, colorDark, ui.fonts.pionillium.medium, ui.anchor.left, ui.anchor.baseline, tooltip)
+
+		-- current distance, relative speed
+		uiPos = ui.pointOnClock(center, radius, 2.5)
+		local distance, distance_unit = ui.Format.Distance(player:DistanceTo(target))
+		local approach_speed = position:dot(velocity) / position:magnitude()
+		local speed, speed_unit = ui.Format.Speed(approach_speed)
+
+		ui.addFancyText(uiPos,
+										{ speed, speed_unit },
+										{ colorLight, colorDark },
+										{ ui.fonts.pionillium.medium, ui.fonts.pionillium.small },
+										ui.anchor.left, ui.anchor.baseline,
+										{ lui.HUD_SPEED_OF_APPROACH_TO_TARGET, lui.HUD_SPEED_OF_APPROACH_TO_TARGET })
+
+		-- current brake distance
+		local brake_distance = player:GetDistanceToZeroV(velocity:magnitude(),"forward")
+		local brake_distance_retro = player:GetDistanceToZeroV(velocity:magnitude(),"reverse")
+		local altitude = player:GetAltitudeRelTo(target)
+		local ratio = brake_distance / altitude
+		local ratio_retro = brake_distance_retro / altitude
+		local speed, speed_unit = ui.Format.Speed(velocity:magnitude())
+
+		-- speed
+		uiPos = ui.pointOnClock(center, radius, 3.5)
+		local distance,unit = ui.Format.Distance(brake_distance)
+		ui.addFancyText(uiPos,
+										{ "~" .. distance, unit },
+										{ colorDark, colorDark },
+										{ ui.fonts.pionillium.medium, ui.fonts.pionillium.small },
+										ui.anchor.left, ui.anchor.baseline,
+										{ lui.HUD_BRAKE_DISTANCE_MAIN_THRUSTERS, lui.HUD_BRAKE_DISTANCE_MAIN_THRUSTERS })
+		-- current altitude
+		uiPos = ui.pointOnClock(center, radius, 3)
+		local altitude, altitude_unit = ui.Format.Distance(altitude)
+		ui.addFancyText(uiPos,
+										{ altitude, altitude_unit, " " .. speed, speed_unit },
+										{ colorLight, colorDark, colorLight, colorDark },
+										{ ui.fonts.pionillium.medium, ui.fonts.pionillium.small, ui.fonts.pionillium.medium, ui.fonts.pionillium.small },
+										ui.anchor.left, ui.anchor.baseline,
+										{ lui.HUD_DISTANCE_TO_SURFACE_OF_TARGET, lui.HUD_DISTANCE_TO_SURFACE_OF_TARGET, lui.HUD_SPEED_RELATIVE_TO_TARGET, lui.HUD_SPEED_RELATIVE_TO_TARGET })
+		-- current speed of approach
+		if approach_speed < 0 then
+			displayReticuleBrakeGauge(center, ratio, ratio_retro)
+		end
+	end
+	-- frame / target switch buttons
+	local uiPos = ui.pointOnClock(center, radius, 4.2)
+	local mouse_position = ui.getMousePos()
+	local size = 24
+	if combatTarget or navTarget then
```

Comment:
See, this is exactly what I'm talking about : from the name of the function, this shouldn't ever register to false, or at least not as late in the function.